settings_dialog.js

```diff
@@ -1,41 +1,55 @@
 // Everything under the settings button
 "use strict";
 
+const fs = require('fs-extra');
```

Comment:
`fs-extra` is being imported to use a single promisified version of a method available in `fs`. I recommend removing `fs-extra` from the package and simply using `util.promisify` to transform the method that is used.